Bottom: 788a763daf85c8079ed9c551e31e4d93aa3d3119
Top:    76977a3542409376ea30b88c35f4d1bac588ff7b
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-10 03:02:02 +0000

More hack to avoid GL calls when we're out of context


---

diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index b8b2e2b..b28f6da 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -506,6 +506,12 @@ ghid_set_color (hidGC gc, const char *name)
 
   gc->colorname = (char *) name;
 
+  if (!check_gl_drawing_ok_hack)
+    {
+      current_color = NULL;
+      return;
+    }
+
   if (gport->colormap == NULL)
     gport->colormap = gtk_widget_get_colormap (gport->top_window);
   if (strcmp (name, "erase") == 0)
